app/main.js

```diff
@@ -95,7 +95,9 @@ app.on('ready', function () {
   });
 
   ipc.on('badge', function(event, arg) {
-    app.dock.setBadge(arg.badge_text);
+    if (process.platform === 'darwin') {
+      app.dock.setBadge(arg.badge_text);
+    }
```

Comment:
You're going to need this for `bounce`, above, if you have enabled that setting. Can I suggest you wrap both `ipc.on('bounce')` and `ipc.on('badge')` in this guard, rather than doing it inside the event?